UnitTests/Tests.cs

```diff
@@ -2518,6 +2544,24 @@ public void CopyToAsyncOneBlock()
             }
         }
 
+        [Test]
+        public void CopyToAsyncOneBlockNonMemoryStream()
+        {
+            using (var stream = GetDefaultStream())
+            {
+                var buffer = GetRandomBuffer(DefaultBlockSize);
+                stream.Write(buffer, 0, buffer.Length);
+                var filename = Path.GetRandomFileName();
+                using (var fileStream = new FileStream(filename, FileMode.Create, FileAccess.ReadWrite, FileShare.ReadWrite, DefaultBlockSize, FileOptions.Asynchronous))
+                {
+                    stream.CopyToAsync(fileStream).Wait();
```

Comment:
Asynchronizing the method could avoid having to do a synchronuos wait here.

Reply:
I saw that, but didn't want to mess with it as part of this PR.

Does NUnit follow the same rule as XUnit and VSTest of Task returning methods for async test methods?

Reply:
Yup, it should just work by having the test method have a `Task`-like return type.

Reply:
Without knowing the threading model of NUnit, I can't make a call here.

If it's not running tests on multiple threads, this might be the most performant way.

Reply:
I generally do not care about the async perf of unit tests, and don't like to get too advanced with NUnit features--keep it simple. I think the Wait is fine here. Correctness is more important.